### ISSUE TITLE
Installation issues with HANA Express 2.0

### ISSUE THREAD

**Issue by rafdre** [SAPDocuments/Tutorials#2036]
I try to install HANA Express since few days and even though I did progress , I think its not over.

The current stage is this:
I started installation with "./setup_hxe.sh"
All went well until that point (only last lines from the log):
```
    Installation of archive file '[/dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSA_CONTENT_10/XSACSERVICES05_3.ZIP]' finished successfully.
  Installing SAPUI5 FESV3 XSA 1 - SAPUI5 SDK 1.44...
    Adding domains...
    Creating or updating services...
    Preparing applications deployment...
    Creating application "sapui5_fesv3"...
    Uploading application "sapui5_fesv3"...
    Installation of the component XSAC_UI5_FESV3 (sap.com) 1.44.23 failed during deployment.
    Installation of archive file(s) '/dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSA_CONTENT_10/XSACUI5FESV344_23.zip' failed.
    Installation of the component XSAC_UI5_FESV3 (sap.com) 1.44.23 failed during deployment.
  Installation of SAPUI5 FESV3 XSA 1 - SAPUI5 SDK 1.44 failed 
Update of SAP HANA Express Edition components failed.
  Installation of XS Advanced components failed
    Installation of SAPUI5 FESV3 XSA 1 - SAPUI5 SDK 1.44 failed

You can send feedback to SAP with this form: https://hana:1129/lmsl/HDBLCM/HXE/feedback/feedback.html
Log file written to '/var/tmp/hdb_HXE_hdblcm_update_2017-11-21_19.36.28/hdblcm.log' on host 'hana'.

Failed to install HANA Extended Services (XSA).
#########################################################################
# Summary after execution 						#
#########################################################################
Server Installation...(OK)
XSC Installation...(OK)
XSA Installation...(FAIL)
HXE Optimization...(NOT STARTED)
Shine Installation...(NOT STARTED)
EA Designer Installation...(NOT STARTED)
Streaming Analytics Installation...(NOT STARTED)
EML Installation...(NOT STARTED)
```
In order to solve the issue I have found log file stored here:
/var/tmp/hdb_HXE_hdblcm_update_2017-11-21_19.36.28/hdblcm.log

here most interesting part of it:

```
20:51:23.834 - INFO:       Output line 55: Preparing applications deployment...
20:51:23.834 - INFO:       Output line 56: Checking application "sapui5_fesv3"...
20:51:29.034 - INFO:       Output line 57: Application "sapui5_fesv3" does not exist
20:51:29.035 - INFO:       Output line 58: Creating application "sapui5_fesv3"...
20:51:29.035 - INFO:       Output line 59: Uploading application "sapui5_fesv3"...
21:07:07.153 - INFO:       Output line 60: Error uploading application "sapui5_fesv3": Could not get files to upload (Scanned 38644 files (629 MB) in 2 m 0 s (5.22 MB/s)): Error executing request PUT https://hana:39030/v2/apps/25defe8b-c410-42c5-9393-4963015e8296/archive/upload/diff?upload_guid=7845f1ca-aaa6-481f-a561-ae4b4646699f: java.lang.InterruptedException
21:07:07.155 - INFO:       Output line 61: Error uploading application "sapui5_fesv3": java.lang.IllegalArgumentException: META-INF/.sap_java_buildpack/tomcat/temp/web7611578545786259534/sapui5/test-resources/sap/suite/ui/commons/demokit/sample/ChartContainerCustomIconsOneChart does not exist
21:07:07.155 - INFO:       Output line 62: Error uploading application "sapui5_fesv3"
21:07:07.155 - INFO:       Output line 63: ---------------- deploy service END --------------------------------
21:07:09.077 - INFO:       Output line 64: Error uploading application "sapui5_fesv3": Could not get files to upload (Scanned 38644 files (629 MB) in 2 m 0 s (5.22 MB/s)): Error executing request PUT https://hana:39030/v2/apps/25defe8b-c410-42c5-9393-4963015e8296/archive/upload/diff?upload_guid=7845f1ca-aaa6-481f-a561-ae4b4646699f: java.lang.InterruptedException
21:07:09.077 - INFO:       Output line 65: Error uploading application "sapui5_fesv3": java.lang.IllegalArgumentException: META-INF/.sap_java_buildpack/tomcat/temp/web7611578545786259534/sapui5/test-resources/sap/suite/ui/commons/demokit/sample/ChartContainerCustomIconsOneChart does not exist
21:07:09.077 - INFO:       Output line 66: Error uploading application "sapui5_fesv3"
21:07:09.077 - INFO:       Output line 67: Error uploading application "sapui5_fesv3": Could not get files to upload (Scanned 38644 files (629 MB) in 2 m 0 s (5.22 MB/s)): Error executing request PUT https://hana:39030/v2/apps/25defe8b-c410-42c5-9393-4963015e8296/archive/upload/diff?upload_guid=7845f1ca-aaa6-481f-a561-ae4b4646699f: java.lang.InterruptedException
21:07:09.077 - INFO:       Output line 68: Error uploading application "sapui5_fesv3": java.lang.IllegalArgumentException: META-INF/.sap_java_buildpack/tomcat/temp/web7611578545786259534/sapui5/test-resources/sap/suite/ui/commons/demokit/sample/ChartContainerCustomIconsOneChart does not exist
21:07:09.077 - INFO:       Output line 69: Error uploading application "sapui5_fesv3"
21:07:09.077 - INFO:       Output line 70: Installation of the component XSAC_UI5_FESV3 (sap.com) 1.44.23 failed during deployment.
21:07:09.514 - INFO:       Output line 71: 
21:07:09.514 - INFO:       Output line 72: Installation of archive file(s) '/dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSA_CONTENT_10/XSACUI5FESV344_23.zip' failed.
```

My solution was to start the command I found in the code:
```
hxeadm@hana:/usr/sap/HXE/HDB90> /hana/shared/HXE/xs/bin/xs install /dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSA_CONTENT_10/XSACUI5FESV344_23.zip -o ALLOW_SC_SAME_VERSION -e /dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSA_CONTENT_10/sap-xsac-shine-1.4.2.mtaext,/dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSAC_SAP_WEB_IDE_20/sap-xsac-devx-4.2.18-hxe.mtaext

Running installation in org "HANAExpress" / space "SAP" as XSA_ADMIN...
File upload starts...
Upload into container starts...
Container ID: 4a89c9ad-ad66-4f26-83ff-4ddf60c4ac0b
                     
Software Component   
-----------------------------------
name                 XSAC_UI5_FESV3
vendor               sap.com
version              1.44.23
-----------------------------------
```

It went fine with most welcome outcome at the end
 ```
---------------- deploy service END --------------------------------
Deployment finished successfully.
Processing of results starts...
Results processed.
Register of software component starts...
Software component XSAC_UI5_FESV3 (sap.com) 'SAPUI5 FESV3 XSA 1 - SAPUI5 SDK 1.44' successfully registered.
Process [install] finished successfully.
Performance /dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSA_CONTENT_10/XSACUI5FESV344_23.zip: UploadFiles 38.3 s (38302 ms), Installation 20 m 24 s (1223613 ms).
Installation of archive file '[/dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSA_CONTENT_10/XSACUI5FESV344_23.zip]' finished successfully.
To see installation logs execute 'xs display-installation-logs 3601 -scv'.
```
Here I could say is ok, however during execution, few times I saw lines like this:
```
Staging application "sapui5_fesv3"...
/bin/sh: mc: line 1: syntax error: unexpected end of file
/bin/sh: error importing function definition for `mc'
bash: mc: line 1: syntax error: unexpected end of file
bash: error importing function definition for `mc'
```

FIRST QUESTION: shall I be concerned about those syntax errors or this is something I can ignore?
SECOND QUESTION: since I resumed the installation process not in any way which is officially described by SAP, do I need to worry that not all parts have been installed successfully ?

What for sure is missing is in my opinion this:
```
HXE Optimization...(NOT STARTED)
Shine Installation...(NOT STARTED)
EA Designer Installation...(NOT STARTED)
Streaming Analytics Installation...(NOT STARTED)
EML Installation...(NOT STARTED)
```
but maybe part of XSA. I am not sure. How can I check it?

**Comment by rafdre** [comment]
I think I have some indication nothing should be missing from XSA, but maybe somebody can confirm.
When I check BASH script I see

```
installXSA() {
	local status=0
	if [ "${COMPONENT}" == "all" ]; then
		echo "Installing HANA Extended Services (XSA)..."
		local log_begin=`wc -l $LOG_FILE | cut -d' ' -f1`
		"${IMAGE_DIR}/${DATA_UNITS_DIR}/${HDB_LCM_COMP_DIR}/hdblcm" -s ${SID} -H ${HOST_NAME} --action=update --components=xs --xs_components=xsac_cockpit,xsac_hrtt,xsac_monitoring,xsac_portal_serv,xsac_sap_web_ide,xsac_services,xsac_ui5_fesv3 --configfile="${IMAGE_DIR}/${DATA_UNITS_DIR}/${HDB_LCM_COMP_DIR}/configurations/auto_install.cfg" --component_medium="${IMAGE_DIR}" --read_password_from_stdin=xml -b <<-EOF
<?xml version="1.0" encoding="UTF-8"?><Passwords><password><![CDATA[${MASTER_PWD}]]></password><system_user_password><![CDATA[${MASTER_PWD}]]></system_user_password><sapadm_password><![CDATA[${MASTER_PWD}]]></sapadm_password><xs_master_password><![CDATA[${MASTER_PWD}]]></xs_master_password><org_manager_password><![CDATA[${MASTER_PWD}]]></org_manager_password><master_password><![CDATA[${MASTER_PWD}]]></master_password></Passwords>
```

it says: xsac_cockpit, xsac_hrtt, xsac_monitoring, xsac_portal_serv, xsac_sap_web_ide, xsac_services, xsac_ui5_fesv3

"xsac_ui5_fesv3" seems to be the last one.

When I check the previous log I see lines:
 ```
Installation of archive file '[/dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSAC_HRTT_20/XSACHRTT02_32.zip]' finished successfully.
(...)
Installation of archive file '[/dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSA_CONTENT_10/XSACMONITORING05_3.ZIP]' finished successfully.
(...)
Installation of archive file '[/dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSA_CONTENT_10/XSACPORTALSERV02_0.zip]' finished successfully.
(...)
Installation of archive file '[/dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSAC_SAP_WEB_IDE_20/XSACSAPWEBIDE02_1.zip]' finished successfully.
(...)
Installation of archive file '[/dane/HANA/HANA_EXPRESS_20/DATA_UNITS/XSA_CONTENT_10/XSACSERVICES05_3.ZIP]' finished successfully.
```

For the other I found bash scripts in installation folder. For those who might need it, here example:

```
hxeadm@hana:/usr/sap/HXE/HDB90> cd /dane/HANA/HANA_EXPRESS_20
hxeadm@hana:/dane/HANA/HANA_EXPRESS_20> ./install_eadesigner.sh
Enter HANA instance number [90]: 
Enter "System database user (SYSTEM)" password : 
Enter "XSA administrator (XSA_ADMIN)" password : 

##############################################################################
# Summary before execution                                                   #
##############################################################################
HANA, express edition installer : /dane/HANA/HANA_EXPRESS_20
  Component(s) to install    : SAP EA Designer
  HANA system ID             : HXE
  HANA instance number       : 90
  System database pasword    : ********
  XSA administrator password : ********
Proceed with installation? (Y/N) : Y
```

This means from the questions I asked the first is still valid plus if somebody can advise me how can I check if installation was done properly it will be great.

**Comment by Lsubatin** [comment]
Hi, the log looks good and that is the last component to be installed.  Right there in /dane/HANA/HANA_EXPRESS_20 you should see a script called hxe_optimize.sh , execute it and you should have the last step covered.
What services do you see when you do xs-admin-login  (use XSA_ADMIN 's password when prompted) and then xs a ?

![image](https://user-images.githubusercontent.com/24896969/33177509-8dd99bc0-d041-11e7-9a2c-b66744545237.png)

Can you log in to webide with xsa_dev? (you'll see the URL there)

**Comment by rafdre** [comment]
hi Lucia, that's what I did yesterday. I executed BASH script "hxe_optimize.sh".
All went well until the last part which I present here:

```
Grant activated role "sap.hana.xs.admin.roles::HTTPDestViewer" to SYSTEM on SystemDB database...
Grant activated role "sap.hana.xs.admin.roles::SQLCCAdministrator" to SYSTEM on SystemDB database...
Grant activated role "sap.hana.xs.debugger::Debugger" to SYSTEM on SystemDB database...
Grant activated role "sap.hana.ide.roles::EditorDeveloper" to SYSTEM on SystemDB database...
Grant activated role "sap.hana.ide.roles::CatalogDeveloper" to SYSTEM on SystemDB database...
Grant activated role "sap.hana.ide.roles::SecurityAdmin" to SYSTEM on SystemDB database...
Grant activated role "sap.hana.ide.roles::TraceViewer" to SYSTEM on SystemDB database...
Grant activated role "sap.hana.xs.admin.roles::HTTPDestViewer" to SYSTEM on HXE database...
Grant activated role "sap.hana.xs.admin.roles::SQLCCAdministrator" to SYSTEM on HXE database...
Grant activated role "sap.hana.xs.debugger::Debugger" to SYSTEM on HXE database...
Grant activated role "sap.hana.ide.roles::EditorDeveloper" to SYSTEM on HXE database...
Grant activated role "sap.hana.ide.roles::CatalogDeveloper" to SYSTEM on HXE database...
Grant activated role "sap.hana.ide.roles::SecurityAdmin" to SYSTEM on HXE database...
Grant activated role "sap.hana.ide.roles::TraceViewer" to SYSTEM on HXE database...
Set system configuration wdisp/system_auto_configuration=true in webdispatcher.ini...
Register SystemDB database HDB resource with Cockpit...
===============================================================================
Register resource on "SystemDB" database
===============================================================================

Login to XSA services...
Check/Wait for Cockpit app to start...

FAILED: Missing apps in space SAP: [cockpit-collection-svc, cockpit-persistence-svc, cockpit-adminui-svc, cockpit-xsa-svc, cockpit-admin-web-app, cockpit-telemetry-svc, cockpit-hdb-svc]

Please start required Cockpit apps and rerun register_cockpit.sh.

Failed to register SystemDB database HDB resource with Cockpit.
Register HXE database HDB resource with Cockpit...
===============================================================================
Register resource on "HXE" database
===============================================================================

Login to XSA services...
Check/Wait for Cockpit app to start...

FAILED: Missing apps in space SAP: [cockpit-collection-svc, cockpit-persistence-svc, cockpit-adminui-svc, cockpit-xsa-svc, cockpit-admin-web-app, cockpit-telemetry-svc, cockpit-hdb-svc]

Please start required Cockpit apps and rerun register_cockpit.sh.

Failed to register HXE database HDB resource with Cockpit.
```

Here is the outcome from "xs a"

```
hxeadm@hana:/usr/sap/HXE/HDB90> xs-admin-login
Enter password for XSA admin user 'XSA_ADMIN': 

API_URL: https://hana:39030
USERNAME: XSA_ADMIN
Authenticating...
ORG: HANAExpress
SPACE: SAP
API endpoint:   https://hana:39030 (API version: 1)
User:           XSA_ADMIN
Org:            HANAExpress
Space:          SAP

hxeadm@hana:/usr/sap/HXE/HDB90> xs a

Getting apps in org "HANAExpress" / space "SAP" as XSA_ADMIN...
Found apps: 
                                                                                    
name                          requested state   instances   memory    disk          urls
------------------------------------------------------------------------------------------------------
auditlog-db                   STOPPED           0/1         16.0 MB   <unlimited>   <none>
auditlog-server               STARTED           1/1         256 MB    <unlimited>   https://hana:51002
auditlog-broker               STARTED           1/1         64.0 MB   <unlimited>   https://hana:51003
deploy-service                STARTED           1/1         280 MB    <unlimited>   https://hana:51004
component-registry-db         STOPPED           0/1         16.0 MB   <unlimited>   <none>
product-installer             STARTED           1/1         256 MB    <unlimited>   https://hana:51005
auditlog-odata                STARTED           1/1         128 MB    <unlimited>   https://hana:51007
auditlog-ui                   STARTED           1/1         64.0 MB   <unlimited>   https://hana:51008
hrtt-service                  STARTED           1/1         512 MB    <unlimited>   https://hana:51009
sqlanlz-svc                   STARTED           1/1         256 MB    <unlimited>   https://hana:51010
sqlanlz-ui                    STARTED           1/1         128 MB    <unlimited>   https://hana:51011
hrtt-core                     STARTED           0/1         512 MB    <unlimited>   https://hana:51012
xsa-admin-backend             STARTED           0/1         128 MB    <unlimited>   https://hana:51016
xsa-admin                     STARTED           0/1         128 MB    <unlimited>   https://hana:51015
sap-portal-static-resources   STARTED           0/1         1.00 GB   <unlimited>   https://hana:51018
sap-portal-persistency        STARTED           0/1         256 MB    <unlimited>   https://hana:51017
di-local-npm-registry         STARTED           0/1         256 MB    <unlimited>   https://hana:51019
di-core                       STARTED           0/1         256 MB    <unlimited>   https://hana:51020
devx-ui5                      STARTED           0/1         128 MB    <unlimited>   https://hana:51021
di-runner                     STARTED           0/1         256 MB    <unlimited>   https://hana:51022
di-cert-admin-ui              STOPPED           0/1         16.0 MB   <unlimited>   https://hana:51023
di-space-enablement-ui        STOPPED           0/1         16.0 MB   <unlimited>   https://hana:51024
javatools                     STARTED           0/1         128 MB    <unlimited>   https://hana:51025
nodejstools                   STARTED           0/1         128 MB    <unlimited>   https://hana:51026
portaltools                   STARTED           0/1         128 MB    <unlimited>   https://hana:51027
webide                        STARTED           0/1         512 MB    <unlimited>   https://hana:53075
jobscheduler-db               STOPPED           0/1         256 MB    <unlimited>   <none>
jobscheduler-rest             STARTED           0/1         256 MB    <unlimited>   https://hana:51030
jobscheduler-service          STARTED           0/1         512 MB    <unlimited>   https://hana:51031
jobscheduler-backend          STARTED           0/1         256 MB    <unlimited>   https://hana:51033
jobscheduler-dashboard        STARTED           0/1         256 MB    <unlimited>   https://hana:51034
jobscheduler-broker           STARTED           0/1         256 MB    <unlimited>   https://hana:51032
sapui5_fesv3                  STARTED           0/1         256 MB    <unlimited>   https://hana:51035
eadesigner-db                 STOPPED           0/1         256 MB    <unlimited>   <none>
eadesigner-backend            STARTED           0/1         1.50 GB   <unlimited>   https://hana:51029
eadesigner                    STARTED           0/1         512 MB    <unlimited>   https://hana:51006
eadesigner-service            STARTED           0/1         512 MB    <unlimited>   https://hana:51028

```
as you can see I also added EA-Designer.

regarding WebIDE, I went in browser to "https://hana:39030/"
there I see link to "https://hana:53075/", but on that page I see message "503 Service Not Available".
maybe I need to wait for all service to start properly since 'XS a' shows me the service on the port 53075 started

**Comment by Lsubatin** [comment]
There are many services not yet started there, so it makes sense to wait. Also, in your manual process you seem not to have installed the cockpit. I personally disable it whenever my instance is just for testing purposes. If you want to install it, you can use (assuming HXE is your SID and "hana" is your hostname):

 hdblcm -s HXE -H hana action=update --components=xs --xs_components=xsac_cockpit --component_medium=<<full path to where the instalation media is>>

Other than that, open web IDE once xs apps shows _STARTED 1/1_

**Comment by rafdre** [comment]
yes. that's what I thought after I posted my last answer.
what I did was "xs restart webide" and now it shows 1/1.
I am able to log in to webide as XSA_DEV

regarding the services : can I decide which start automatically and which I prefer to start manually using 'xs start SERVICE' ? For instance "EA Designer" I want to have it, but maybe will not use it all the time.

regarding the command you provided: something wrong with it. see this:

```
hxeadm@hana:/usr/sap/HXE/HDB90> hdblcm -s HXE -H hana action=update --components=xs --xs_components=xsac_cockpit --component_medium=<>
-bash: syntax error near unexpected token `newline'
```

I found a hint here: [https://stackoverflow.com/questions/5134399/bash-syntax-error-near-unexpected-token-newline](url)
When I remove <> I get different error:

```
hxeadm@hana:/usr/sap/HXE/HDB90> hdblcm -s HXE -H hana action=update --components=xs --xs_components=xsac_cockpit --component_medium=
If 'hdblcm' is not a typo you can use command-not-found to lookup the package that contains it, like this:
    cnf hdblcm
```

**Comment by rafdre** [comment]
ok. I found this command here: /hana/shared/HXE/hdblcm

```
SAP HANA Lifecycle Management - SAP HANA Express Edition 2.00.021.00.1507025078
*******************************************************************************
The following actions cannot be selected because the tool was not started as root user:
    - Unregister the SAP HANA Express Edition System
    - Rename the SAP HANA Express Edition System

Choose an action

  Index | Action             | Description
  ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  1     | check_installation | Check SAP HANA Express Edition Installation
  2     | extract_components | Extract Components
  3     | uninstall          | Uninstall SAP HANA Express Edition Components
  4     | update_components  | Install or Update Additional Components
        |                    |   Update is pending (since 2017-11-21T21:07:59+0100)
        |                    |   SAPUI5 FESV3 XSA 1 - SAPUI5 SDK 1.44: Install of version 1.044.23 is pending (since 2017-11-21T20:47:55) at step 'Install SAPUI5 FESV3 XSA 1 - SAPUI5 SDK 1.44'
  5     | update_host        | Update the SAP HANA Express Edition Instance Host integration
  6     | exit               | Exit (do nothing)

```

**Comment by Lsubatin** [comment]
I love it when people reply to themselves :P 

You can go into the xsa_admin cockpit ( https://hana:51015 ) and stop the services you do not want to start automatically from there. Or use  _xs stop <<app name>>_ from the CLI.

**Comment by rafdre** [comment]
I do not want them to stop. I want prevent system from starting them automatically.  This is different.

First thing I noticed is: HANA Express starts automatically with the system, so I stopped it in profile settings by setting "Autostart=0". I am looking for something similar to define which apps start automatically after I execute 'HDB start'. For instance I want to start "eadesigner" only when I need it.

Is there any option for it? In xsa_admin cockpit  I have not found this option.

**Comment by Lsubatin** [comment]
Not that I know of. If you stop them before you reboot, they will not start automatically when you start XSA again. This whole thread, this belongs in the community Q&A (issues are for the tutorials themselves, not for installation problems or general questions), so I'm closing this.

**Comment by syedhaider79** [comment]
Hi

I installed SAP Hana Express on IBM Cloud and it stuck at a point. Before starting VM i updated RAM to 32GB and Processor 4 x 2. The disk is HDD not SSD. It timed out during start of service. I restarted all XS APP stopped services including mentioned in the image.

Want to restart cockpit but anyone can guide what action parameter should i use ?

Any help will be appreciated. 
 
I logged into the console 
![error-stuck](https://user-images.githubusercontent.com/31449627/36467404-0ee78af4-16f8-11e8-8c21-d906cd38db7b.JPG)
